zone/aggro.cpp

```diff
@@ -745,10 +745,27 @@ bool Mob::IsAttackAllowed(Mob *target, bool isSpellAttack)
 	// can't damage own pet (applies to everthing)
 	Mob *target_owner = target->GetOwner();
 	Mob *our_owner = GetOwner();
-	if(target_owner && target_owner == this)
+	Mob* target_ultimate_owner = target->GetUltimateOwner();
```

Comment:
This is a relatively hot path, `GetUltimateOwner` hits the entity list which makes it expensive. We should fetch this only when `IsBot()` is true, not under every time we call this. 

`target_ultimate_owner` and `our_ultimate_owner` should only be fetched when needed (Under IsBot())